lib/ansible/plugins/action/eos.py

```diff
@@ -43,7 +43,7 @@ def run(self, tmp=None, task_vars=None):
 
         socket_path = None
 
-        if self._play_context.connection == 'network_cli':
+        if self._play_context.connection in ('network_cli', 'httpapi'):
```

Comment:
@Qalthos Do we want to add this check in action/net_base also?

Reply:
It isn't necessary, and there isn't really an analogue to this check in net_base now, so I don't think so.